src/flowcean/environments/rosbag.py

```diff
@@ -0,0 +1,84 @@
+from os import listdir
+from pathlib import Path
+from typing import Self, override
+
+import polars as pl
+from rosbags.dataframe import get_dataframe
+from rosbags.highlevel import AnyReader
+from rosbags.typesys import Stores, get_types_from_msg, get_typestore
+
+from flowcean.core.environment import OfflineEnvironment
+
+
+def guess_msgtype(path: Path) -> str:
+    """Guess message type name from path."""
+    name = path.relative_to(path.parents[2]).with_suffix("")
+    if "msg" not in name.parts:
+        name = name.parent / "msg" / name.name
+    return str(name)
+
+
+class RosbagEnvironment(OfflineEnvironment):
+    """Environment for rosbags."""
+
+    def __init__(
+        self,
+        path: str | Path,
+        topics: dict[str, list[str]],
+        custom_msgs_path: str | Path | None = None,
+    ) -> None:
+        """Initialize the RosbagEnvironment.
+
+        Args:
+            path: Path to the rosbag.
+            topics: Dictionary of topics to load. The keys are the topic names
+                and the values are lists of keys inside each message that
+                shall be extracted.
```

Comment:
I know what you mean, but that feels really difficult to grasp and use for a normal enduser. Can we find another, clearer way to express the same? Also it should probably be `field` instead of `keys`.